### PR TITLE
Adding .travis.yml file

### PR DESCRIPTION
**Pull request by nbessi**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#    - ODOO="https://github.com/odoo/odoo/archive/master.tar.gz"
+#    - ODOO="https://github.com/OCA/OCB/archive/master.zip"
+
+before_install:
+    - git clone https://github.com/OCA/partner-contact-management $HOME/partner-contact-management
+    - cd $HOME/partner-contact-management && git checkout 7.0
+    - git clone https://github.com/OCA/bank-statement-reconcile $HOME/bank-statement-reconcile
+    - cd $HOME/bank-statement-reconcile && git checkout 7.0
+    - git clone https://github.com/OCA/account-financial-tools $HOME/account-financial-tools
+    - cd $HOME/account-financial-tools && git checkout 7.0
```
Comment (bwrsandman) [inline .travis.yml]: You can use

``` bash
git clone https://github.com/OCA/account-financial-tools $HOME/account-financial-tools -b 7.0
```

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Config file .travis.yml
+
+language: python
+
+python:
+    - "2.7"
+
+
+env:
+    - ODOO="https://github.com/savoirfairelinux/odoo/archive/setuptools-addons.tar.gz"  # Temp until https://github.com/odoo/odoo/issues/185 or https://github.com/odoo/odoo/issues/441 is fixed
```
Comment (bwrsandman) [inline .travis.yml]: FYI, this is a 8.0 server branch

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# Test with flake, ignore F401 for __init__.py files, use a max length of 120
+# Run tests with coverage
+# Only test modules in repo (list populated by directories in repo)
+# Preload modules before testing to only run tests of repo's modules
+# Include current directory and dependent repos in addons-path as well as official addons
+script:
+# We will activate flake 8 in a dedicated commit
+#    - flake8 . --max-line-length=120 --filename=__init__.py --ignore=F401
+#    - flake8 . --max-line-length=120 --exclude=__init__.py
+    - openerp.py -d test --stop-after-init --init=$(python -c 'import os; print(",".join(x for x in os.listdir(".") if os.path.isdir(x) and not x.startswith(".") and not x.endswith(".unported")))') --addons-path=$(pwd),`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/addons,$HOME/partner-contact-management,$HOME/bank-statement-reconcile,$HOME/account-financial-tools
```
Comment (bwrsandman) [inline .travis.yml]: openerp_server

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+    - coverage run $(which odoo.py) -d test --test-enable --log-level=test --stop-after-init --init=$(python -c 'import os; print(",".join(x for x in os.listdir(".") if os.path.isdir(x) and not x.startswith(".") and not x.endswith("_unported")))') --addons-path=$(pwd),`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/addons
```
Comment (bwrsandman) [inline .travis.yml]: coverage run $(which openerp_server)

Comment (nbessi) [inline .travis.yml]: Thanks I was looking for this

**Comment by nbessi** [comment]
I think a git rebase will be welcome :wink:

File: .travis.yml (hunk 4 above)
Comment (bwrsandman) [inline .travis.yml]: I did this work yesterday for 6.1, but gave up as, I thought 7.0, 6.1 and 6.0 branches were to be readonly on github and dev to be done on launchpad.

https://github.com/savoirfairelinux/canada/blob/b07761002554f077163d992b68384c58c57080db/.travis.yml

**Comment by bwrsandman** [comment]
I did similar work yesterday for 6.1, but gave up as, I thought 7.0, 6.1 and 6.0 branches were to be readonly on github and dev to be done on launchpad.

https://github.com/savoirfairelinux/canada/blob/b07761002554f077163d992b68384c58c57080db/.travis.yml

**Comment by bwrsandman** [comment]
> I think a git rebase will be welcome :wink:

Oh my! I thought you were doing `git commit --amend`

**Comment by nbessi** [comment]
@bwrsandman, did not think about this option to retrigger Travis build :dizzy_face:

**Comment by bwrsandman** [comment]
Yeah travis triggers on a commit hash change. Pretty neat.

**Comment by bwrsandman** [comment]
We probably need to do a similar patch to https://github.com/odoo/odoo/issues/185 for setup.py for the version 7.0

**Comment by nbessi** [comment]
Tho I'm looking for the addons path of addons

**Comment by nbessi** [comment]
Finally all tests are launched correctly :metal: :metal:

**Comment by nbessi** [comment]
But that sucks I have error in tests but return status is 0 :scream:

**Comment by nbessi** [comment]
@bwrsandman should we encapsulate the run with a grep on Error ??

**Comment by bwrsandman** [comment]
@nbessi typically, it's `start_openerp ... | tee logfile.txt && grep ERROR logfile`

I personally don't like that at all.

Alternatively, assign a log file to grep to.

Maybe test failing should return -1?

**Comment by nbessi** [comment]
Yes it is better idea

**Comment by bwrsandman** [comment]
Looking at Odoo code, it will always return 0 on exit except in the case of uncaught errors.

This is not a very good way of doing tests.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+# Include current directory and dependent repos in addons-path as well as official addons
+script:
+# We will activate flake 8 in a dedicated commit
+#    - flake8 . --max-line-length=120 --filename=__init__.py --ignore=F401
+#    - flake8 . --max-line-length=120 --exclude=__init__.py
+    - openerp-server -d test --stop-after-init --init=$(python -c 'import os; print(",".join(x for x in os.listdir(".") if os.path.isdir(x) and not x.startswith(".") and not x.endswith(".unported")))') --addons-path=$(pwd),`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/addons,$HOME/partner-contact-management,$HOME/bank-statement-reconcile,$HOME/account-financial-tools --logfile=/tmp/test.log.txt; cat /tmp/test.log.txt; if [ `grep -i error /tmp/test.log.txt` ]; then exit 1; else exit 0; fi
+
+    - coverage run $(which openerp-server) -d test --test-enable --log-level=test --stop-after-init --init=$(python -c 'import os; print(",".join(x for x in os.listdir(".") if os.path.isdir(x) and not x.startswith(".") and not x.endswith("_unported")))') --addons-path=$(pwd),`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/addons
```
Comment (bwrsandman) [inline .travis.yml]: Tests are done in this line

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+# Test with flake, ignore F401 for __init__.py files, use a max length of 120
+# Run tests with coverage
+# Only test modules in repo (list populated by directories in repo)
+# Preload modules before testing to only run tests of repo's modules
+# Include current directory and dependent repos in addons-path as well as official addons
+script:
+# We will activate flake 8 in a dedicated commit
+#    - flake8 . --max-line-length=120 --filename=__init__.py --ignore=F401
+#    - flake8 . --max-line-length=120 --exclude=__init__.py
+    - openerp-server -d test --stop-after-init --init=$(python -c 'import os; print(",".join(x for x in os.listdir(".") if os.path.isdir(x) and not x.startswith(".") and not x.endswith(".unported")))') --addons-path=$(pwd),`python -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())"`/addons,$HOME/partner-contact-management,$HOME/bank-statement-reconcile,$HOME/account-financial-tools --logfile=/tmp/test.log.txt; cat /tmp/test.log.txt; if [ `grep -i error /tmp/test.log.txt` ]; then exit 1; else exit 0; fi
```
Comment (bwrsandman) [inline .travis.yml]: This is a pre-load script, no tests are done here.

**Comment by nbessi** [comment]
@bwrsandman the same is true for master ??

**Comment by bwrsandman** [comment]
At first glance it would seem not
8.0 https://github.com/odoo/odoo/blob/master/openerp/cli/server.py#L169
vs
7.0 https://github.com/odoo/odoo/blob/7.0/openerp/cli/server.py#L265

**Comment by lepistone** [comment]
Grepping for "ERROR" has the problem that a passing assertRaises does log "ERROR". It would be better to grep for the message that says at the end that all tests passed (or failed).

**Comment by nbessi** [comment]
Test failed but they failed correctly :)

**Pull request merged**